### ISSUE TITLE
Newer versions of task do not work with older versions of Octopus

### ISSUE THREAD

**Issue by mattfinner** [OctopusDeploy/OctoTFS#239]
Hi,

We're currently using v3 of "OctopusPush" and "OctopusCreateRelease" tasks in our azdo yaml release pipelines. Are getting a warning that these versions have been deprecated.

Tried to upgrade to both v5 and v4, but it seems the "Spaces" input is mandatory. We are running an older version of Octo server (3.17.14) which does not support Spaces.

Obviously a bit worried that v3 tasks are deprecated and we can't upgrade to newer versions. Are there any plans to drop v3 tasks and if so, is there any chance of a work around for older versions of Octo?

Cheers,
Matt

**Comment by jbristowe** [comment]
Hey @mattfinner! 👋 Thank you for submitting the issue and bringing this issue to our attention.

**TL/DR: v3 tasks are NOT going away. We should not have marked them as deprecated. We will fix this.**

Your assessment of the current situation is correct. For the purpose of context, here is a summary of the situation:

- You are running Octopus Server 3.17.14, which does not support Spaces
- Spaces were introduced in Octopus Server 2019.1, which shipped on January 28, 2019
- v3 of the `OctopusPush` and `OctopusCreateRelease` tasks in OctoTFS 5.0.109 are marked as `[DEPRECATED]` in the default name and listed as "deprecated" in its description
- The Spaces pick list is a `required` input field for all v4+ tasks, except for `OctoCli`, `OctoInstaller`, and `OctopusPack`

In retrospect, we used the wrong wording with v3 and v4 tasks when we shipped OctoTFS 5.0.57. We are in the process of updating this because of the feedback we received from customers like you. Again, thank you for bringing this to our attention.

Recently, we merged a PR to update the wording for v4 tasks (#238). This issue correctly identifies the need for us to update the wording for v3 tasks as well.

Rest assured, we do not have plans to drop v3 tasks that ship with OctoTFS.
